### ISSUE TITLE
Regression: glitches and black screens on scene demos and Death Rally

### ISSUE THREAD

**Issue by digistorm79** [joncampbell123/dosbox-x#3561]
### Describe the bug

I noticed a major regression in the macOS ARM64 build 0.84.0, compared to 0.83.23.
(I don't compare to 0.83.24 because that crashes DOSBox itself on a lot of occasions)

When running on the dynamic core, a couple of scene demos major graphical problems. Some demos just show a black screen or lock-up on launch. For example:
- The demo [Sunflower](https://www.pouet.net/prod.php?which=85) has just black screen, and after a while DOSBox crashes itself.
- The intro [Jizz](https://www.pouet.net/prod.php?which=2) crashes with black screen
- The demo [My summer demo](https://www.pouet.net/prod.php?which=71415) just has a black screen.
- The intro [Gamma](https://www.pouet.net/prod.php?which=2516) has just black screen and will not quit on escape key.
- The intro [Jive 2](https://www.pouet.net/prod.php?which=2195) stalls on first screen
- The demo [Riprap](https://www.pouet.net/prod.php?which=306) crashes on start screen
- The demo [Machines of madness](https://www.pouet.net/prod.php?which=312) crashes on load.
- The demo [Solstice](https://www.pouet.net/prod.php?which=106) just displays black screen and music runs way too slow.

[Death Rally shareware version](https://www.dosgamesarchive.com/download/death-rally/) has yellow background glitch in the main menu, crashes when the game should start.

Some scene demos glitch or even freak out completely. For example:
- The demo [Shattered Minds](https://www.pouet.net/prod.php?which=77397) spams “argument singularity in log2” on the screen and then completely borks and the timing seems completely broken.
- The demo [Goldilocks](https://www.pouet.net/prod.php?which=67125) has weird palette errors
- The demo [303](https://www.pouet.net/prod.php?which=44) just glitches weird stuff, music plays even more too slow than it does on normal core (which is a different bug).
- The demo [Headache](https://www.pouet.net/prod.php?which=1150) just glitches weird. Some effects work, some don't.

All examples were working fine in version 0.83.23 Apple M1 build.

The Intel build of version 0.84.0 does not have these errors.
The normal core of the ARM64 build of version 0.84.0 also does not have these errors.

Some demos also suffer from a timing bug, but because that bug also manifests on the normal cpu core and also on the Intel build, I made a separate bug report of that.

### Steps to reproduce the behaviour

Download the mentioned demo or game.
Configure appropriately (if needed).
Run (most examples are demos, you can just run them)
Compare with version 0.83.23.
It is very obvious.

### Expected behavior

On the pouet pages there are YouTube captures of the expected behaviour. In addition, since this is a regression, in version 0.83.23 things were just okay.

### What operating system(s) this bug have occurred on?

macOS Monterey

### What version(s) of DOSBox-X have this bug?

0.84.0

### Additional information

All tests were performed on a M1 Pro Macbook Pro.

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by digistorm79** [comment]
I want to raise this issue again. It has been several versions, and this bug, which is (largely) a regression, is still not fixed, making DOSBox-X unreliable to use on arm64 architecture, because anything remotely new will not work. It is especially frustrating that this particular bug was not present in version 0.83.23.
I already asked in the vanilla DOSBox forum about this bug, and one user responded: "This seems to be a DOSBox-X issue. I tried it on Linux aarch64, and it behaves just as you described. Vanilla DOSBox works."
So is it maybe please possible to revert whatever change was made to the dynamic core? It is better to have a slow dynamic core than a broken one.

**Comment by joncampbell123** [comment]
We've made some fixes to dynamic core in the last month, especially FPU emulation, that may have resolved this issue.

I am able to run Windows 98 and Quake on my ARM Macbook just fine, so far.

**Comment by digistorm79** [comment]
Yes, all software mentioned is working now, and it apparently fixed a lot of other issue too. That's great.
